prisma/schema.prisma

```diff
@@ -17,30 +17,31 @@ datasource db {
 model Dept {
   id           String   @id @default(uuid())
   name         String
-  abbreviation String
+  abbreviation String   @unique
   courses      Course[]
 }
 
 model Course {
   id     String @id @default(uuid())
-  name   String
+  name   String?
   code   String
   term   String
-  year   Int
   dept   Dept   @relation(fields: [deptId], references: [id])
   deptId String
-  books  Book[]
+  books  Book[]   @relation("_CourseBookRelationTable")
+
+  @@unique([code, deptId], name: "courseIdentifier")
 }
 
 model Book {
   id                String   @id @default(uuid())
-  isbn              String
+  isbn              String   @unique
   name              String
   imageUrl          String?
   googleBooksId     String?
   isCampusStoreBook Boolean
   campusStorePrice  Int?
-  courses           Course[]
+  courses           Course[]   @relation("_CourseBookRelationTable")
```

Comment:
Did not work before, so I added this.

Reply:
Really? The Prisma documentation says it should work https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations

What issue were you running in to?

Reply:
I got this error saying it could not find the relation table between them:
<img width="758" alt="Screen Shot 2022-01-18 at 2 11 09 PM" src="https://user-images.githubusercontent.com/40670885/150002981-d12a724c-43a6-4b56-96d1-2a77a1b73751.png">

Reply:
Running the following works fine with no errors after removing the explicit relation and running the migration against my DB:
```typescript
const book = await prisma.book.create({
  data: {
    isCampusStoreBook: true,
    isbn: "123",
    name: "test",
    courses: {
      create: {
        code: "123",
        name: "course_name",
        dept: {
          create: {
            abbreviation: "123",
            name: "test",
          },
        },
        term: "term",
      },
    },
  },
});

console.log(book);

console.log(await prisma.book.findFirst({ include: { courses: true } }));
console.log(await prisma.course.findFirst({ include: { books: true } }));
```

Output:
```
> book-bazar@ index /Users/calebmech/Developer/Book-Bazar
> ts-node ./bookstore-scrapper/indexTextbooks.ts

{
  id: 'aa7e3edc-9f6b-4e82-9c99-523d0e518a14',
  isbn: '123',
  name: 'test',
  imageUrl: null,
  googleBooksId: null,
  isCampusStoreBook: true,
  campusStorePrice: null
}
{
  id: 'aa7e3edc-9f6b-4e82-9c99-523d0e518a14',
  isbn: '123',
  name: 'test',
  imageUrl: null,
  googleBooksId: null,
  isCampusStoreBook: true,
  campusStorePrice: null,
  courses: [
    {
      id: 'd292d15b-e7ff-4647-b0af-ab8346e3ea4c',
      name: 'course_name',
      code: '123',
      term: 'term',
      deptId: '1d4cc796-1bdc-4c6c-9c7c-f7facb70c8a8'
    }
  ]
}
{
  id: 'd292d15b-e7ff-4647-b0af-ab8346e3ea4c',
  name: 'course_name',
  code: '123',
  term: 'term',
  deptId: '1d4cc796-1bdc-4c6c-9c7c-f7facb70c8a8',
  books: [
    {
      id: 'aa7e3edc-9f6b-4e82-9c99-523d0e518a14',
      isbn: '123',
      name: 'test',
      imageUrl: null,
      googleBooksId: null,
      isCampusStoreBook: true,
      campusStorePrice: null
    }
  ]
}
```

Reply:
I also just noticed you have not committed the migration for your database change. Did you just forget to commit it?

https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate

You may need to start by reseting your database from scratch by running `npm run prisma migrate reset` to get things into a good state.

Reply:
Started working after doing some of the migration steps. Thanks Caleb.